### ISSUE TITLE
Access Violation when trying to save as anything but png or jpg

### ISSUE THREAD

**Issue by agallero** [mono/SkiaSharp#64]
Trying the following code raises and AV:
```
    class Program
    {
        static void Main(string[] args)
        {
            SKBitmap bmp = new SKBitmap(10, 10);
            IntPtr len;
            SKImage img = SKImage.FromPixels(bmp.Info, bmp.GetPixels(out len), bmp.RowBytes);
            SKData data = img.Encode(SKImageEncodeFormat.Gif, 10);
            using (FileStream fs = new FileStream("..\\..\\test.gif", FileMode.Create))
            {
                data.SaveTo(fs);
            }

        }
    }

```

I am not sure if I am doing somehting wrong in the code, but it works with png and jpeg. The others fail with and AV.

**Comment by mattleibow** [comment]
Ok, the issue is that the `SKImage.Encode` is returning a `NULL` in native code, but is not properly reflected in the managed code.

```c

    /**
     *  Encode the image's pixels and return the result as a new SkData, which
     *  the caller must manage (i.e. call unref() when they are done).
     *
     *  If the image type cannot be encoded, or the requested encoder type is
     *  not supported, this will return NULL.
     *
     *  Note: this will attempt to encode the image's pixels in the specified format,
     *  even if the image returns a data from refEncoded(). That data will be ignored.
     */
```

**Comment by mattleibow** [comment]
The currently supported encoders are:
 - KTX
 - JPEG
 - PNG
 - WEBP

**Comment by mattleibow** [comment]
"Fixed" in 733264b7be888c215da7bb204658a9ed0314ad38

Although not a fix for supporting the other encoders, the fix correctly returns `null` if the encoder couldn't be created. The new exception will be the correct `NullReferenceException`.
